config/configload/load.go

```diff
@@ -285,203 +141,41 @@ func LoadConfig(body *hclsyntax.Body) (*config.Couper, error) {
-			jwtSigningConfigs[jwt.Name] = signConf
-		}
+	jwtSigningConfigs, e := helper.configureJWTSigningConfig()
+	if e != nil {
+		return nil, e
 	}
 
 	helper.config.Context = helper.config.Context.(*eval.Context).
 		WithJWTSigningConfigs(jwtSigningConfigs).
 		WithOAuth2AC(helper.config.Definitions.OAuth2AC).
 		WithSAML(helper.config.Definitions.SAML)
 
-	definedACs := make(map[string]struct{})
-	for _, ac := range helper.config.Definitions.BasicAuth {
-		definedACs[ac.Name] = struct{}{}
-	}
-	for _, ac := range helper.config.Definitions.JWT {
-		definedACs[ac.Name] = struct{}{}
-	}
-	for _, ac := range helper.config.Definitions.OAuth2AC {
-		definedACs[ac.Name] = struct{}{}
-	}
-	for _, ac := range helper.config.Definitions.OIDC {
-		definedACs[ac.Name] = struct{}{}
-	}
-	for _, ac := range helper.config.Definitions.SAML {
-		definedACs[ac.Name] = struct{}{}
-	}
-
-	// Read per server block and merge backend settings which results in a final server configuration.
-	for _, serverBlock := range hclbody.BlocksOfType(body, server) {
-		serverConfig := &config.Server{}
-		if diags := gohcl.DecodeBody(serverBlock.Body, helper.context, serverConfig); diags.HasErrors() {
-			return nil, diags
-		}
-
-		// Set the server name since gohcl.DecodeBody decoded the body and not the block.
-		if len(serverBlock.Labels) > 0 {
-			serverConfig.Name = serverBlock.Labels[0]
-		}
-
-		if err := checkReferencedAccessControls(serverBlock.Body, serverConfig.AccessControl, serverConfig.DisableAccessControl, definedACs); err != nil {
-			return nil, err
-		}
-
-		for _, fileConfig := range serverConfig.Files {
-			if err := checkReferencedAccessControls(fileConfig.HCLBody(), fileConfig.AccessControl, fileConfig.DisableAccessControl, definedACs); err != nil {
-				return nil, err
-			}
-		}
-
-		for _, spaConfig := range serverConfig.SPAs {
-			if err := checkReferencedAccessControls(spaConfig.HCLBody(), spaConfig.AccessControl, spaConfig.DisableAccessControl, definedACs); err != nil {
-				return nil, err
-			}
-		}
-
-		// Read api blocks and merge backends with server and definitions backends.
-		for _, apiConfig := range serverConfig.APIs {
-			apiBody := apiConfig.HCLBody()
-
-			if apiConfig.AllowedMethods != nil && len(apiConfig.AllowedMethods) > 0 {
-				if err = validMethods(apiConfig.AllowedMethods, apiBody.Attributes["allowed_methods"]); err != nil {
-					return nil, err
-				}
-			}
-
-			if err := checkReferencedAccessControls(apiBody, apiConfig.AccessControl, apiConfig.DisableAccessControl, definedACs); err != nil {
-				return nil, err
-			}
-
-			rp := apiBody.Attributes["required_permission"]
-			if rp != nil {
-				apiConfig.RequiredPermission = rp.Expr
-			}
-
-			err = refineEndpoints(helper, apiConfig.Endpoints, true, definedACs)
-			if err != nil {
-				return nil, err
-			}
-
-			err = checkPermissionMixedConfig(apiConfig)
-			if err != nil {
-				return nil, err
-			}
+	helper.defsACs = getDefinedACs(helper.config.Definitions)
```

Comment:
getDefinedACs() uses helper.config.Definitions, so it should have a *helper receiver, like the other moved functions:
```go
func (h *helper) getDefinedACs() map[string]struct{} {
	definitions := h.config.Definitions # or use h.config.Definitions.* in ranges
...
```

As defsACs is only used in the scope of helper.configureServers() and helper.configureAPIs() (called from within helper.configureServers()), I would make it a local variable in helper.configureServers() and pass it to the using functions:
```go
func (h *helper) configureServers(body *hclsyntax.Body) error {
	var err error

	defsACs := h.getDefinedACs()
...
		err = h.configureAPIs(serverConfig.APIs, defsACs)
...
}

func (h *helper) configureAPIs(apis config.APIs, defsACs map[string]struct{}) error {
```